bluej/src/main/java/bluej/stride/generic/FrameCursor.java

```diff
@@ -402,11 +402,27 @@ public FrameCursor(final InteractionManager editor, final FrameCanvas parentCanv
                 Frame target = parentCanvas.getFrameBefore(FrameCursor.this);
                 if (target != null)
                 {
+                    final boolean isDeletingJointFrame = (target instanceof MultiCanvasFrame) && target.getCanvases().count() > 1;
                     editor.beginRecordingState(FrameCursor.this);
-                    FrameCursor cursorBeforeTarget = parentCanvas.getCursorBefore(target);
+                    FrameCursor cursorBeforeTarget = (isDeletingJointFrame) ? FrameCursor.this : parentCanvas.getCursorBefore(target);
                     editor.recordEdits(StrideEditReason.FLUSH);
-                    editor.showUndoDeleteBanner(target.calculateEffort());
-                    parentCanvas.removeBlock(target);
+                    final int deleteEffort;
+                    if(isDeletingJointFrame)
+                    {
+                        // In the case of backspace being hit below a "joint" frame (that is, a multicanvas frame with
+                        // more elements than only the root), we do not delete the complete frame but only remove the last joint.
+                        final FrameCanvas lastJointCanvas =  ((MultiCanvasFrame)target).getLastCanvas();
+                        deleteEffort = lastJointCanvas.getBlockContents().stream()
+                            .map((frame) -> frame.calculateEffort())
+                            .reduce(0, (subTotal, effort) -> subTotal + effort);
```

Comment:
For info, a slightly simpler/neater way to do this with Java streams is to use mapToInt on the first call, and then that gives you an IntStream which has a .sum() method.